airbyte-integrations/connectors/source-microsoft-teams/source_microsoft_teams/manifest.yaml

```diff
@@ -0,0 +1,5801 @@
+            type:
+              - 'null'
+              - string
+          usedAndroidPhone:
+            type:
+              - 'null'
+              - string
+          usedChromeOS:
+            type:
+              - 'null'
+              - string
+          usedIOs:
+            type:
+              - 'null'
+              - string
+          usedLinux:
+            type:
+              - 'null'
+              - string
+          usedMac:
+            type:
+              - 'null'
+              - string
+          usedWeb:
+            type:
+              - 'null'
+              - string
+          usedWindows:
+            type:
+              - 'null'
+              - string
+          usedWindowsPhone:
+            type:
+              - 'null'
+              - string
+          userId:
+            type:
+              - 'null'
+              - string
+          userPrincipalName:
+            type:
+              - 'null'
+              - string
+        type:
+          - 'null'
+          - object
+    retriever:
+      type: SimpleRetriever
+      requester:
+        type: HttpRequester
+        url_base: https://graph.microsoft.com/v1.0/
+        path: >-
+          reports/getTeamsDeviceUsageUserDetail(period='{{ config['period']
+          }}')?$top={{ config['page_size'] }}
+        http_method: GET
+        request_parameters: {}
+        request_headers: {}
+        authenticator:
+          type: OAuthAuthenticator
+          client_id: '{{ config[''client_id''] }}'
+          client_secret: '{{ config[''client_secret''] }}'
+          refresh_request_body:
+            scope: https://graph.microsoft.com/.default
+          token_refresh_endpoint: >-
+            https://login.microsoftonline.com/{{ config['tenant_id']
+            }}/oauth2/v2.0/token
+          grant_type: client_credentials
+          scopes: []
+          access_token_name: access_token
+        error_handler:
+          type: CompositeErrorHandler
+          error_handlers:
+            - type: DefaultErrorHandler
+              response_filters:
+                - type: HttpResponseFilter
+                  action: IGNORE
+                  http_codes:
+                    - 400
+                  error_message: Personal account not supported
+        request_body_json: {}
+      record_selector:
+        type: RecordSelector
+        extractor:
+          type: DpathExtractor
+          field_path:
+            - value
+      paginator:
+        type: DefaultPaginator
+        page_token_option:
+          type: RequestPath
+        pagination_strategy:
+          type: CursorPagination
+          cursor_value: '{{ response.get("@odata.nextLink", {}) }}'
+          stop_condition: '{{ not response.get("@odata.nextLink", {}) }}'
+spec:
+  connection_specification:
+    $schema: http://json-schema.org/draft-07/schema#
+    type: object
+    required:
+      - client_id
+      - client_secret
+      - period
+      - tenant_id
+    properties:
```

Comment:
let's move these properties in a `credentials` block so it's compatible with the existing spec

Reply:
Done, It's backward compatible now